### ISSUE TITLE
few commits to review

### ISSUE THREAD

**Issue by sthalik** [skullernet/q2pro#126]
Hey,

Here are few commits you may want to cherry-pick:

[note, post was edited once right after sending]

- https://github.com/sthalik/q2pro/commit/2a019d5d7e01ff9576c7f29489b20198f522160c - the evil `timeGetTime()` returns the same time value unless the process was preempted. This may be bad.
- https://github.com/sthalik/q2pro/commit/d70566bb86acc8f54ec6a885af3f503ec329facc - add missing header, prototype, fix dinput build (raw input still works better).
- https://github.com/sthalik/q2pro/commit/dba63d66d12f319d4b3b8c8ccc55203ed0d53f51 - "fix" modelist truncation on big displays, and actually show line that expanded indefinitely. This helps with all kinds of broken aliases.
- https://github.com/sthalik/q2pro/commit/4f7d9daabf0437ea2273a21f0ffe401e332f4ae8 - add texturemode where trilinear mapping is used as min filter, while NEAREST as a magnify filter. Old-school pixellated textures with distant geometry looking fine.

**Comment by skullernet** [comment]
>sthalik/q2pro@2a019d5

Nice, but if timeGetTime() call gets removed timeBeginPeriod() and timeEndPeriod() should be removed too. And QueryPerformanceFrequency() call can be moved to Sys_Init() I think.

>sthalik/q2pro@d70566b

platform.h include in not really needed, there is unwritten rule that shared.h is always included first, which pulls platform.h. Z_LeakTestAll() seems unrelated. dinput.c fix is OK but also has unrelated whitespace changes.

>sthalik/q2pro@dba63d6

I don't think MAX_STRING_CHARS can be blindly increased without breaking client/server/game compatibility.

>sthalik/q2pro@4f7d9da 

OK.

**Comment by sthalik** [comment]
I opened separate pull requests for each commit. Haven't seen your review first, will accompany changes. Oops.

It should be easier to debug using github's review functionality though.

> platform.h include in not really needed, there is unwritten rule that shared.h is always included first, which pulls platform.h. Z_LeakTestAll() seems unrelated. dinput.c fix is OK but also has unrelated whitespace changes.

Otherwise the header fails to parse basic syntax in IDEs when viewing the file itself.

`Z_LeakTestAll()` is useful for e.g. debugging biggun crash. I grouped it with prototypes for other `Z_`-prefix functions.

I'm backing off the dinput changes. It's more jerky than raw input. Also it's supported on Windows XP so there's not really a need to improve upon it IMO.

In fact I recommend removing dinput support entirely. Getting it right with event buffering is hard. For a long time I had lost joystick/key keydown events.

> I don't think MAX_STRING_CHARS can be blindly increased without breaking client/server/game compatibility.

Will look into it in the newly-opened PRs, as well as your other comments.
